src/components/pod/pod.component.js

```diff
@@ -26,328 +25,220 @@ const marginPropTypes = filterStyledSystemMarginProps(
   styledSystemPropTypes.space
 );
 
-class Pod extends React.Component {
-  state = {
-    isEditHovered: false,
-    isEditFocused: false,
-    isDeleteHovered: false,
-    isDeleteFocused: false,
-    isUndoHovered: false,
-    isUndoFocused: false,
-  };
-
-  static contextType = PodContext;
-
-  handleEditMouseEnter = () => {
-    this.setState({ isEditHovered: true });
-  };
-
-  handleEditMouseLeave = () => {
-    this.setState({ isEditHovered: false });
-  };
-
-  handleEditFocus = () => {
-    this.setState({ isEditFocused: true });
-  };
+const Pod = forwardRef(
+  (
+    {
+      alignTitle = "left",
+      border = true,
+      children,
+      className,
+      displayEditButtonOnHover,
+      editContentFullWidth,
+      footer,
+      height,
+      internalEditButton,
+      onDelete,
+      onEdit,
+      onUndo,
+      size = "medium",
+      softDelete,
+      subtitle,
+      title,
+      triggerEditOnContent,
+      variant = "primary",
+      ...rest
+    },
+    ref
+  ) => {
+    const [isEditFocused, setEditFocused] = useState(false);
+    const [isEditHovered, setEditHovered] = useState(false);
+    const [isDeleteFocused, setDeleteFocused] = useState(false);
+    const [isDeleteHovered, setDeleteHovered] = useState(false);
+    const [isUndoFocused, setUndoFocused] = useState(false);
+    const [isUndoHovered, setUndoHovered] = useState(false);
 
-  handleEditBlur = () => {
-    this.setState({ isEditFocused: false });
-  };
+    let podHeight;
 
-  handleDeleteMouseEnter = () => {
-    this.setState({ isDeleteHovered: true });
-  };
+    if (height && typeof height === "number") {
+      podHeight = `${height}px`;
+    } else if (height) {
+      podHeight = height;
+    }
```

Comment:
I guess we can do something like 
```js
if(height){
  podHeight = `${String(height)px}`
}
```
so I think that we won't need conditionals here and then maybe that useRef won't be necessary